string what
bool type
---
bool success